### PR TITLE
SPI support in bootstub

### PR DESCRIPTION
**Pull request by adeebshihadeh**

### CODE REVIEW

File: board/stm32fx/peripherals.h [hunk 1]
```diff
@@ -5,6 +5,15 @@ void gpio_usb_init(void) {
   GPIOA->OSPEEDR = GPIO_OSPEEDER_OSPEEDR11 | GPIO_OSPEEDER_OSPEEDR12;
 }
 
+void gpio_spi_init(void) {
```
Comment (robbederks) [inline board/stm32fx/peripherals.h:8]: This might be more board-specific, why not just have this in `dos` and `red_chiplet`?

**Review by robbederks — APPROVED** [review]

File: board/stm32fx/peripherals.h (hunk 1 above)
Comment (adeebshihadeh) [reply board/stm32fx/peripherals.h:8]: because we don’t do the whole board init in the flasher, but perhaps that’s a legacy thing?

Comment (robbederks) [reply board/stm32fx/peripherals.h:8]: Ah, I see.
Should be fine to keep it as-is then

File: python/__init__.py [hunk 2]
```diff
@@ -440,7 +449,7 @@ def flash_static(handle, code, mcu_type):
     # erase sectors
     logging.warning(f"flash: erasing sectors 1 - {last_sector}")
     for i in range(1, last_sector + 1):
-      handle.controlWrite(Panda.REQUEST_IN, 0xb2, i, 0, b'')
+      handle.controlWrite(Panda.REQUEST_IN, 0xb2, i, 0, b'', timeout=int(15*1e4))
```
Comment (robbederks) [inline python/__init__.py:452]: why 1e4? libusb's timeout is in ms, which would need 1e3

File: python/spi.py [hunk 3]
```diff
@@ -96,9 +96,10 @@ def _calc_checksum(self, data: List[int]) -> int:
       cksum ^= b
     return cksum
 
-  def _wait_for_ack(self, spi, ack_val: int) -> None:
+  def _wait_for_ack(self, spi, timeout: int, ack_val: int) -> None:
     start = time.monotonic()
-    while (time.monotonic() - start) < ACK_TIMEOUT_SECONDS:
+    timeout_seconds = max(timeout * 1e-4, MIN_ACK_TIMEOUT_SECONDS)
```
Comment (robbederks) [inline python/spi.py:101]: same here, 1e3

**Pull request merged**